File: Tests/Unit/RAnalyticsEndpointSettableTests.m
#import <Kiwi/Kiwi.h>
#import <AdSupport/AdSupport.h>
#import <RAnalytics/RAnalyticsEndpointSettable.h>

@interface RAnalyticsEndpointHandler : NSObject<RAnalyticsEndpointSettable>
@end
@implementation RAnalyticsEndpointHandler
@synthesize endpointURL;
@end

SPEC_BEGIN(RAnalyticsEndpointSettableTests)

describe(@"RAnalyticsEndpointSettable", ^{
    describe(@"endpoint", ^{
        it(@"should return https//endpoint.com when endpoint equals https//endpoint.com", ^{
            RAnalyticsEndpointHandler *analyticsEndpointHandler = RAnalyticsEndpointHandler.new;
            NSURL * _Nonnull endpoint = [NSURL URLWithString:@"https//endpoint.com"];
            analyticsEndpointHandler.endpointURL = endpoint;
            [[analyticsEndpointHandler.endpointURL should] equal:endpoint];
            [[analyticsEndpointHandler.endpointURL.absoluteString should] equal:endpoint.absoluteString];
        });
    });
});

SPEC_END

File: UnitTestApp/Tests/Unit/RAnalyticsEndpointSettableTests.swift
import Testing
import Foundation
@testable import RakutenAnalytics
#if canImport(RAnalyticsTestHelpers)
import RAnalyticsTestHelpers
#endif

private final class RAnalyticsEndpointHandler: NSObject, EndpointSettable {
    var endpointURL: URL?
}

@Suite("RAnalyticsEndpointSettable")
struct RAnalyticsEndpointSettableTests {
    @Suite("endpointURL")
    struct EndpointURLTests {
        @Test("should return https//endpoint.com when the endpoint is https//endpoint.com")
        func testShouldReturnHttpsEndpointComWhenTheEndpointIsHttpsEndpointCom() {
            let endpoint = URL(string: "https//endpoint.com")
            let analyticsEndpointHandler = RAnalyticsEndpointHandler()
            analyticsEndpointHandler.endpointURL = endpoint
            #expect(analyticsEndpointHandler.endpointURL == endpoint)
            #expect(analyticsEndpointHandler.endpointURL?.absoluteString == "https//endpoint.com")
        }
    }
}
